src/components/Footer.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { Todo } from '../types/Todo';
+
+interface FooterProps {
+  todos: Todo[];
+  activeFilter: string;
+  setActiveFilter: React.Dispatch<React.SetStateAction<string>>;
+  handleClearCompleted: () => void;
+}
+
+const Footer: React.FC<FooterProps> = ({
+  todos,
+  activeFilter,
+  setActiveFilter,
+  handleClearCompleted,
+}) => {
+  if (todos.length === 0) {
+    return null;
+  }
+
+  const notCompletedCount = todos.filter(todo => !todo.completed).length;
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {notCompletedCount} items left
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={`filter__link ${activeFilter === 'All' ? 'selected' : ''}`}
+          data-cy="FilterLinkAll"
+          onClick={() => setActiveFilter('All')}
+        >
+          All
+        </a>
+        <a
+          href="#/active"
+          className={`filter__link ${activeFilter === 'Active' ? 'selected' : ''}`}
+          data-cy="FilterLinkActive"
+          onClick={() => setActiveFilter('Active')}
+        >
+          Active
+        </a>
+        <a
+          href="#/completed"
+          className={`filter__link ${activeFilter === 'Completed' ? 'selected' : ''}`}
+          data-cy="FilterLinkCompleted"
+          onClick={() => setActiveFilter('Completed')}
+        >
+          Completed
```

Comment:
Use `Object.values(your created enum)` and render these options with `map()` method